conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/data/Instantiables.java

```diff
@@ -32,18 +30,14 @@ final class Instantiables {
 
     private Instantiables() {}
 
+    static CodeBlock instantiate(TypeMirror mirror) {
+        return instantiate(MoreTypes.asDeclared(mirror));
+    }
+
     /**
      * Produces a {@link CodeBlock} of instantiation code for the provided type.
      * This supports both no-arg class constructors and single-value enum singletons.
      */
-    static CodeBlock instantiate(TypeMirror mirror) {
-        DeclaredType declaredType = MoreTypes.asDeclared(mirror);
-        if (declaredType == null) {
-            throw new SafeIllegalStateException("TypeMirror is not a DeclaredType", SafeArg.of("type", mirror));
-        }
```

Comment:
`MoreTypes.asDeclared` throws an `IllegalArgumentException` if the given `TypeMirror` is not a `DeclaredType`. It will never return `null`.